### PR TITLE
Converted jigsaw apollo test

### PR DESCRIPTION
**Pull request by krlberry**
## Description
Converts the jigsaw apollo test to gtest. 

Also includes:
* A helper function to compare csv lines for convenience.
* Updates to the `Cube::createFromIsd()` functions to include exception throws if the isd or pvl file cannot be opened. This 
was an issue I had and had difficulty debugging without this change because the error I received (for this issue) was a "json parse" exception.

## Related Issue
Part of test conversion.

## Motivation and Context
Part of test conversion issue. Decreases size of necessary test data. 

## How Has This Been Tested?
This is a test. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsJigsaw.cpp]: Probably move this to Utilities.cpp?

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsJigsaw.cpp]: probably get rid of the "1" and just call it ui

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsJigsaw.cpp]: this path should probably be saved as part of the fixture.

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsJigsaw.cpp]: This process is definitely better for comparing headers than what I was doing. That being said, as jigsaw is kinda all over the place (in the sense of it relies on a number of objects). Try to check what parts of the CSV isn't already tested by one of these objects as part of a unittest (I think bundleobservation and bundleresults do a lot of the work? there is also bundleUtilities/unittest.cpp that does a lot of testing).

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsJigsaw.cpp:183]: I wonder how necessary these EXPECT_THROW are? If nothing is catching them, can we not let them naturally fail?

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsJigsaw.cpp]: I recall you having issues with the test data area but this should probably be a relative path or a path in fixture (as Jenkins doesn't seem to complain with relative paths).

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsJigsaw.cpp]: Also should probably go to utilities.cpp.

File: isis/src/base/objs/Cube/Cube.cpp
Comment (Kelvinrr) [inline isis/src/base/objs/Cube/Cube.cpp:140]: this is good because I would sometimes mess up the path to the ISD and I get a bogus error from the JSON lib. Would this also trigger in the case of a file not existing? (presumably yes).

**Review by Kelvinrr — COMMENTED** [review]
some questions in line. I would favor this over the other Apollo PR with only the fixtures so I vote for closing the other.

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (krlberry) [reply isis/tests/FunctionalTestsJigsaw.cpp]: Good catch, didn't realize I'd left this in.

File: isis/src/base/objs/Cube/Cube.cpp
Comment (krlberry) [reply isis/src/base/objs/Cube/Cube.cpp:140]: Yeah, it also triggers if the file doesn't exist, which was the issue I was running into!

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (krlberry) [reply isis/tests/FunctionalTestsJigsaw.cpp:183]: My intent here was to indicate that the this part for the test resulting in an exception would be a failure of the test, like _explicitly_, i.e. "not throwing is something being tested" and this was my attempt to indicate that.  

There would be no technical problem with just letting them fail, and I'm not sure how much extra this adds. Maybe I'll add one of those "custom error messages" to supply more context.

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (krlberry) [reply isis/tests/FunctionalTestsJigsaw.cpp]: Can do. Didn't do initially because I wasn't sure how useful they would be elsewhere.

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (krlberry) [reply isis/tests/FunctionalTestsJigsaw.cpp:183]: I did this. Let me know what you think.

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (krlberry) [reply isis/tests/FunctionalTestsJigsaw.cpp]: Good point. 

I did some grepping, and it looks like the header values for the any of the csv files aren't being tested anywhere, so I think it makes sense to leave the header comparisons in for each of the output csv files, too, and just test them in this _one_ jigsaw test. I don't think the headers need to be tested for any other jigsaw tests, since they're covered here.

The other csv line comparisons just "spot-check" a few points to make sure their exact values match the expected values in the current output, though I have no particular preference for doing this as opposed to taking the statics approach you took. I'm tempted to leave all of this in. What do you think?

**Comment by krlberry** [comment]
I'm going to wait until the apollo fixture is merged (PR #4080) and then go ahead and merge dev into this branch to fix merge conflicts.

**Review by Kelvinrr — DISMISSED** [review]
if jenkins comes back okay I think this is good to merge

File: isis/tests/FunctionalTestsJigsaw.cpp
Comment (krlberry) [inline isis/tests/FunctionalTestsJigsaw.cpp]: @jessemapel This was my attempt to compare the corrected latitude and longitude in meters from the control network to the csv output, but the comparison I did here didn't quite work out. (See: 1000.0 as the tolerance.) Do you know what the appropriate numbers to compare would be and if they are even available at the ControlNetwork level? (The information that is written directly to the csv uses the correction from the bundle directly.)

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**
